### ISSUE TITLE
Arithmatex: removing dollar inline syntax gives "IndexError: no such group"

### ISSUE THREAD

**Issue by pxeger** [facelessuser/pymdown-extensions#1564]
### Description

When configuring Arithmatex to disable the `$` syntax for inline mathematics with this configuration:
```yaml
# ...
  - pymdownx.arithmatex:
      # ...
      inline_syntax: [round]
```
Any markdown which contains `\( \)` (which is the`round` syntax left enabled) will fail to render with this exception:

```
Traceback (most recent call last):
  ...
  File ".../site-packages/markdown/core.py", line 268, in convert
    newRoot = treeprocessor.run(root)
  File ".../site-packages/markdown/treeprocessors.py", line 371, in run
    self.__handleInline(text), child
  File ".../site-packages/markdown/treeprocessors.py", line 131, in __handleInline
    data, matched, startIndex = self.__applyPattern(
  File ".../site-packages/markdown/treeprocessors.py", line 272, in __applyPattern
    node, start, end = pattern.handleMatch(match, data)
  File ".../site-packages/pymdownx/arithmatex.py", line 254, in handleMatch
    escapes = m.group(4)
IndexError: no such group
```

All of the following settings for `inline_syntax` still work:
```yaml
[round, dollar] # this is the default anyway
[dollar]
[]
```
So it's just this combination of round only that breaks.

This issue also appears to be unrelated to any of the other Arithmatex options (e.g. `smart_dollar: false` does not fix it)

### Minimal Reproduction

1. Configure pymdown-extensions to enable Arithmatex and set `inline_syntax: [round]`
2. Try to render the markdown `\( 1 + 2 = 3 \)`
3. Observe the error

### Version(s) & System Info

- Operating System: Arch Linux
- Python Version: 3.10.1
- Package Version: 9.1

**Comment by facelessuser** [comment]
Looks like it breaks with just `dollar` for me as well when using a dollar equation instead of a round bracketed equation.

**Comment by facelessuser** [comment]
I've located what the issue is and applied a patch. I assume, when the patterns could not be limited to either `$...$` or `\(...\)`, this always worked. Whenever we made this configurable, we forgot to put in some tests to ensure we didn't break functionality. At some point, we did break functionality.

While I have a fix in place, I want to go back in and put some tests as well. I also will want to double-check that block math doesn't fall victim to similar bad logic.

Regardless, the next release should have this properly resolved when it releases. Thanks for finding this and filing an issue!
